src/new/sendTransaction.ts

```diff
@@ -188,22 +190,33 @@ class SendTransaction extends EventEmitter {
       });
     }
 
-    const partialTxData: {inputs: IDataInput[], outputs: IDataOutput[]} = {inputs: [], outputs: []};
+    const partialTxData: Pick<IDataTx, "outputs" | "inputs"> = {inputs: [], outputs: []};
     for (const [token, chooseInputs] of tokenMap) {
-      const options: IFillTxOptions = { chooseInputs, skipAuthorities: true };
+      const options: IUtxoSelectionOptions = {
+        token,
+        chooseInputs,
+      };
       if (this.changeAddress) {
         options.changeAddress = this.changeAddress;
       }
-      const newData = await this.storage.fillTx(token, txData, options);
-      partialTxData.inputs.push(...newData.inputs);
-      partialTxData.outputs.push(...newData.outputs);
+      const proposedData = await prepareSendTokensData(this.storage, txData, options);
+      partialTxData.inputs.push(...proposedData.inputs);
+      partialTxData.outputs.push(...proposedData.outputs);
+    }
+
+    let outputs: IDataOutput[];
+    if (partialTxData.outputs.length === 0) {
+      outputs = txData.outputs;
+    } else {
+      // Shuffle outputs, so we don't have change output always in the same index
+      outputs = shuffle([...txData.outputs, ...partialTxData.outputs])
```

Comment:
I'm not sure, the idea of shuffling is so that we can prevent someone from knowing 2 addresses are from the same wallet by identifying an output as change.

So not having the change makes the shuffle not really necessary